iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -275,36 +275,74 @@ typedef struct DEVICE_METHOD_INFO_TAG
     STRING_HANDLE request_id;
 } DEVICE_METHOD_INFO;
 
-static void free_proxy_data(MQTTTRANSPORT_HANDLE_DATA* mqtt_transport_instance)
+//
+// InternStrnicmp implements strnicmp.  strnicmp isn't available on all platforms.
+//
+static int InternStrnicmp(const char* s1, const char* s2, size_t n)
+{
+    int result;
+
+    if (s1 == NULL) result = -1;
+    else if (s2 == NULL) result = 1;
```

Comment:
nit: not sure we allow as coding style one line `if` or `else if`

Reply:
Fixed.